### PR TITLE
ddtrace/tracer: add support for DD_TRACE_ANALYTICS_ENABLED env. var.

### PR DESCRIPTION
**Pull request by gbbr**
This change adds support for the DD_TRACE_ANALYTICS_ENABLED environment
variable, which will enable App Analytics for web integrations.

### CODE REVIEW

File: ddtrace/tracer/option.go [hunk 1]
```diff
@@ -102,6 +104,9 @@ func newConfig(opts ...StartOption) *config {
 	}
 	c.dogstatsdAddr = net.JoinHostPort(statsdHost, statsdPort)
 
+	if v := os.Getenv("DD_TRACE_ANALYTICS_ENABLED"); v != "" {
```
Comment (knusbaum) [inline ddtrace/tracer/option.go:107]: Are we sure we want to accept any value as turning it on?
I think `DD_TRACE_ANALYTICS_ENABLED=false` causing it to be enabled is counter-intuitive.

Maybe we can parse it as a boolean.

File: ddtrace/tracer/option.go [hunk 2]
```diff
@@ -89,6 +89,8 @@ type config struct {
 // StartOption represents a function that can be provided as a parameter to Start.
 type StartOption func(*config)
 
+// newConfig renders the tracer configuration based on defaults, environment variables
+// and passed user opts.
```
Comment (knusbaum) [inline ddtrace/tracer/option.go:93]: 🙇

File: ddtrace/tracer/option.go (hunk 1 above)
Comment (gbbr) [reply ddtrace/tracer/option.go:107]: You’re right. It was a hasty copy pasta. I should use the `boolEnv` you have in your PR.

Comment (knusbaum) [reply ddtrace/tracer/option.go:108]: `boolEnv` https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/tracer/option.go#L497 is merged now, if you want to use it. It's not critical, though.

**Pull request merged**